### PR TITLE
fix miscellaneous bugs related to new resource hierarchy migration

### PR DESCRIPTION
**Pull request by wangmingliang-ms**
[AB#1931889](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1931889): [Test] Should disable Straming Logs/Profile Flight Recorder/Edit Deployment for deleted function/webapp/deployment in favorite list.
[AB#1931543](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1931543): [Test]Failed to "start streaming log" operation on slot of web app.
[AB#1931509](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1931509): [Test] IDE error "java.lang.Throwable: Write-unsafe context" when create VM after reopen cretion dialog
[AB#1931234](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1931234): [Test] Can open properties view for a deleted resource in favorite list
[AB#1931546](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1931546): [Test]Existing app service plans are not listed when creating a web app/function

### CODE REVIEW

File: Utils/azure-toolkit-ide-libs/azure-toolkit-ide-springcloud-lib/src/main/java/com/microsoft/azure/toolkit/ide/springcloud/SpringCloudActionsContributor.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ public void registerActions(AzureActionManager am) {
         final Consumer<SpringCloudApp> openTestUrl = s -> am.getAction(ResourceCommonActionsContributor.OPEN_URL).handle(s.getTestUrl());
         final ActionView.Builder openTestUrlView = new ActionView.Builder("Access Test Endpoint", "/icons/action/browser.svg")
                 .title(s -> Optional.ofNullable(s).map(r -> title("springcloud.open_test_url.app", ((SpringCloudApp) r).name())).orElse(null))
-                .enabled(s -> s instanceof SpringCloudApp && ((SpringCloudApp) s).exists());
+                .enabled(s -> s instanceof SpringCloudApp && !((SpringCloudApp) s).isDraftForCreating());
```
Comment (wangmingliang-ms) [inline Utils/azure-toolkit-ide-libs/azure-toolkit-ide-springcloud-lib/src/main/java/com/microsoft/azure/toolkit/ide/springcloud/SpringCloudActionsContributor.java]: should be `!isConnected()`

File: Utils/azure-toolkit-ide-libs/azure-toolkit-ide-appservice-lib/src/main/java/com/microsoft/azure/toolkit/ide/appservice/webapp/WebAppNodeProvider.java [hunk 2]
```diff
@@ -76,7 +77,10 @@ public boolean accept(@Nonnull Object data, @Nullable Node<?> parent) {
 
     @Nullable
     public static AzureIcon.Modifier getOperatingSystemModifier(AppServiceAppBase<?, ?, ?> resource) {
-        final OperatingSystem os = Optional.ofNullable(resource.getRuntime()).map(r -> r.getOperatingSystem()).orElse(null);
-        return resource.getFormalStatus().isWaiting() ? null : os != OperatingSystem.WINDOWS ? AzureIcon.Modifier.LINUX : null;
+        if (resource.getFormalStatus().isWaiting()) {
```
Comment (wangmingliang-ms) [inline Utils/azure-toolkit-ide-libs/azure-toolkit-ide-appservice-lib/src/main/java/com/microsoft/azure/toolkit/ide/appservice/webapp/WebAppNodeProvider.java:80]: return `null` directly if `isDeleted()`.

File: Utils/azure-toolkit-ide-libs/azure-toolkit-ide-appservice-lib/src/main/java/com/microsoft/azure/toolkit/ide/appservice/webapp/WebAppNodeProvider.java [hunk 3]
```diff
@@ -79,6 +79,8 @@ public boolean accept(@Nonnull Object data, @Nullable Node<?> parent) {
     public static AzureIcon.Modifier getOperatingSystemModifier(AppServiceAppBase<?, ?, ?> resource) {
         if (resource.getFormalStatus().isWaiting()) {
             return null;
+        } else if (!resource.getFormalStatus().isConnected()) {
+            return null;
```
Comment (wangmingliang-ms) [inline Utils/azure-toolkit-ide-libs/azure-toolkit-ide-appservice-lib/src/main/java/com/microsoft/azure/toolkit/ide/appservice/webapp/WebAppNodeProvider.java:81]: merge these 2 `if`s

**Review by Flanker32 — DISMISSED** [review]
LGTM

**Pull request merged**